packages/@aws-cdk/aws-ec2/test/vpc.test.ts

```diff
@@ -1573,24 +1553,15 @@ describe('vpc', () => {
       });
 
       // THEN - No exception
-      expect(stack).toHaveResource('Some::Resource', {
+      Template.fromStack(stack).hasResourceProperties('Some::Resource', {
         subnetIds: { 'Fn::Split': [',', { 'Fn::ImportValue': 'myPublicSubnetIds' }] },
       });
 
-      // THEN - Warnings have been added to the stack metadata
-      const asm = SynthUtils.synthesize(stack);
-      expect(asm.messages).toEqual(expect.arrayContaining([
-        expect.objectContaining(
-          {
-            entry: {
-              type: 'aws:cdk:warning',
-              data: "fromVpcAttributes: 'availabilityZones' is a list token: the imported VPC will not work with constructs that require a list of subnets at synthesis time. Use 'Vpc.fromLookup()' or 'Fn.importListValue' instead.",
-            },
-          },
-        ),
-      ]));
-
-
+      expect(vpc.node.metadataEntry).toContainEqual({
```

Comment:
@kaizen3031593 this looks like the solution to what we were talking about yesterday.

Reply:
Good call. Although I am about to write a PR for messages support that would look something like this:

```ts
AssertAnnotations.fromStack(stack).hasWarning('*', {
  entry: {
    type: 'aws:cdk:warning',
    data: "fromVpcAttributes...",
  },
});
```